src/mpi/mpi_caf.c

```diff
@@ -758,10 +757,12 @@ PREFIX (init) (int *argc, char ***argv)
         caf_runtime_error ("MPI_THREAD_MULTIPLE is not supported: %d", prov_lev);
 #else
       if (is_init) {
-          caf_owns_mpi = false;
+	caf_owns_mpi = false;
       } else {
-          MPI_Init (argc, argv);
-          caf_owns_mpi = true;
+	MPI_Init_thread (argc, argv, prior_thread_level, &prov_lev);
+	caf_owns_mpi = true;
+	if (caf_this_image == 0 && MPI_THREAD_FUNNELED != prov_lev)
+	  caf_runtime_error ("MPI_THREAD_FUNNELED is not supported: %d", prov_lev);
```

Comment:
@afanfa is there a way to write a test to trigger this runtime error?

I don't completely understand what `prov_lev` does and how it's set.